src/poseidon/poseidon_circuit.rs

```diff
@@ -394,14 +396,20 @@ impl<F: PrimeField + PrimeFieldBits, const T: usize, const RATE: usize> Poseidon
 
 #[cfg(test)]
 mod tests {
+    use halo2_proofs::{
+        circuit::{Layouter, SimpleFloorPlanner},
+        plonk::{Circuit, Column, ConstraintSystem, Instance},
+    };
+    use halo2curves::{
+        group::ff::FromUniformBytes,
+        pasta::{EqAffine, Fp},
+    };
+
+    use crate::{
+        create_and_verify_proof, main_gate::MainGateConfig, poseidon::Spec, run_mock_prover_test,
+    };
+
     use super::*;
```

Comment:
I arrange the imports according to the rules
- First the standard library `std::`
- Then all external crates.
- Then `crate::`
- Then relative `super::`.

cargo-fmt can do this, but only in nightly (I've been waiting for stabilization for a few years now), but it helps to orientate imports faster.

Reply:
That is, I make these edits by Boy Scout rule if I affect module imports, but since these rules are not automatically enabled - I do not add them to our cargo-fmt pipeline

Anyway, I did it for a reason.